src/runtime/naming/name.cpp

```diff
@@ -239,6 +241,35 @@ namespace hpx { namespace naming
 
         ///////////////////////////////////////////////////////////////////////
         // prepare the given id, note: this function modifies the passed id
+        void handle_credit_splitting(
+            serialization::output_archive& ar, id_type_impl& gid,
+            serialization::detail::preprocess_gid_types* split_gids)
+        {
+            auto* handle_futures = ar.try_get_extra_data<
```

Comment:
I am not very confident with those changes here. I'd rather like to have the original semantics restored, as in, assume we always have a future handler and always have a split gid handler. Let's work on other use cases as soon as we need them.

Reply:
Under normal circumstances (i.e with `output_archives` constructed without specifying extra data) should we always rely on credit-splitting and/or future handling to be available? Or should we remove the ability to construct `output_archives` without extra data in the first place?

My reasoning was that users normally don't need credit-splitting or handling of (HPX) futures. They simply want to construct an archive and have their serialization needs covered.

Reply:
That's correct, but in our parcel layer, we know that those might need to be serialized. We absolutely need to handle them. So in our case, we can just add them when constructing the archive (we need them anyway), let's try to move on with this with as little extra changes, and deal with additional usage requirements along the lines.

Reply:
That would either require to a) touch on all examples and tests that use serialization directly or b) have any `output_archive` that is constructed without explicit extra data support all of the internal HPX extra data items. Both options don't look appealing to me.